pkg/query-service/app/queryBuilder/query_builder.go

```diff
@@ -322,13 +322,12 @@ func isLogExpression(expression *govaluate.EvaluableExpression, params *v3.Query
 func (c *cacheKeyGenerator) GenerateKeys(params *v3.QueryRangeParamsV3) map[string]string {
 	keys := make(map[string]string)
 
-	// For non-graph panels, we don't support caching
-	if params.CompositeQuery.PanelType != v3.PanelTypeGraph {
-		return keys
-	}
-
```

Comment:
This is the main change. We used to return empty keys for every non-graph panel type. We still do it, but only when

1. query type is promql
2. The data source is logs

Now when it comes to metrics, things change

1. v3 - same as logs, only graph queries can be cached
2. v4 - everything can be cached

Reply:
We can't enable caching for other panels for logs and traces due to how their queries work. Unlike metrics, log queries don't return time series data that can be easily aggregated in query service. Instead, they fetch a single aggregated value for the entire time window. This makes it impossible to cache results for future use, like p95, avg, etc. We can support caching for simpler operations like sum or count, but we'd need to implement that first before enabling it here.

Reply:
Also there's no v4 for logs or traces - they work the same way.

Reply:
ok